crates/evmexec/src/engine.rs

```diff
@@ -529,8 +529,8 @@ mod tests {
 
         let timestamp = 0;
         let el_ops = vec![];
-        let safe_l1_block = Buf32(FixedBytes::<32>::random());
-        let prev_l2_block = Buf32(FixedBytes::<32>::random()).into();
+        let safe_l1_block = FixedBytes::<32>::random().into();
+        let prev_l2_block = Buf32(FixedBytes::<32>::random().into()).into();
```

Comment:
Could we get rid of this extra `Buf32` here too?

Reply:
we can, but then `L2BlockId` (in the crate `state`) would need an additional `From` impl, I guess.

Given it's in the tests only (I didn't see this pattern in the code, but maybe I overlooked somewhere), I'd leave it as is, if you don't mind.

Reply:
I think adding a `From<Buf32>` makes sense there.

Would it make sense add another macro that can instantiate the impls for types that are just wrappers around hashes like `L2BlockId` and `L1BlockId`?

Reply:
Good idea one another macro, done.

> I think adding a `From<Buf32>` makes sense there.

@delbonis Sorry, did you mean `From<FixedBytes>`? `From<Buf32>` is already there. 
I now think adding `From<FixedBytes>` actually makes little sense. One obvious drawback - it adds a dependency on `reth_pritimitives` for the `strata-state` crate. Right now, `strata-state` depends only on `strata-primitives` which I find ergonomic.

Nonetheless, not a strong opinion, let me know what do you think, so I can wrap this piece of work soon :)

Reply:
Oh then I guess we're all set.  Yeah we want to avoid extra deps on `reth_primitives`.